indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,366 @@
+import PropTypes from 'prop-types';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {Form as FinalForm} from 'react-final-form';
+import Leaflet from 'leaflet';
+import {FeatureGroup, Map, Rectangle, TileLayer, Tooltip, ZoomControl} from 'react-leaflet';
+import {EditControl} from 'react-leaflet-draw';
+import {Button, Form, Header, Modal} from 'semantic-ui-react';
+
+import {FinalCheckbox, FinalInput, FinalSubmitButton} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+
+import {selectors as mapSelectors, actions as mapActions} from '../../common/map';
+import {selectors as configSelectors} from '../../common/config';
+
+import 'leaflet/dist/leaflet.css';
+import 'leaflet-draw/dist/leaflet.draw.css';
+import './MapAreasPage.module.scss';
+
+
+Leaflet.EditToolbar.Delete.include({
+    removeAllLayers: false
+});
+
+const DEFAULT_CENTER = [46.233832398, 6.053166454];
+
+class MapAreasPage extends React.Component {
+    static propTypes = {
+        tileServerURL: PropTypes.string.isRequired,
+        areas: PropTypes.arrayOf(PropTypes.object).isRequired,
+        actions: PropTypes.exact({
+            createMapArea: PropTypes.func.isRequired,
+            updateMapAreas: PropTypes.func.isRequired,
+            deleteMapAreas: PropTypes.func.isRequired,
+            fetchMapAreas: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.map = null;
+        this.featureGroup = null;
+    }
+
+    state = {
+        areaProps: null,
+        suppressClick: false
+    };
+
+    componentDidMount() {
+        const {actions: {fetchMapAreas}} = this.props;
+        fetchMapAreas();
+        // rerender is required here because only at this point this.map
+        // refers to the actual map element which is then necessary in this.drawAreas
+        this.forceUpdate();
+    }
+
+    setMapAreaRef = ref => {
+        this.map = ref;
+    };
+
+    setFeatureGroupRef = ref => {
+        this.featureGroup = ref;
+    };
+
+    createNewArea = (evt) => {
+        const {
+            layerType,
+            layer: {
+                _bounds: {_northEast: northEast, _southWest: southWest}
+            }
+        } = evt;
+
+        if (layerType !== 'rectangle') {
+            return;
+        }
+
+        this.setState({
+            areaProps: {bounds: {northEast, southWest}}
+        });
+
+        if (this.featureGroup) {
+            // this is necessary here so that we don't end up with
+            // the initial layer being still visible on the map
+            this.featureGroup.leafletElement.eachLayer((layer) => {
+                const {options: {id}} = layer;
+                if (id === undefined) {
+                    this.featureGroup.leafletElement.removeLayer(layer);
+                }
+            });
+        }
+    };
+
+    editArea = (evt) => {
+        if (!Object.keys(evt.layers._layers).length) {
```

Comment:
Any particular reason why this is before the `const` declarations?

Reply:
Why would you put it here? I don't need those definitions at this point.

Reply:
I know, just code style. IIRC some linters have this rule to have definitions on top.